### PR TITLE
[MAINTENANCE] databricks_sql integration tests

### PR DESCRIPTION
**Pull request by Kilo59**
Enable integration tests with databricks sql.

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [x] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: tests/datasource/fluent/integration/test_sql_datasources.py [hunk 1]
```diff
@@ -242,51 +269,63 @@ def test_postgres(
...
             engine=snowflake_ds.get_engine(),
             table_names={table_name},
-            schema=get_random_identifier_name(),
+            schema=schema,
         )
 
-        table_names: list[str] = inspect(snowflake_ds.get_engine()).get_table_names()
+        table_names: list[str] = get_engine_inspector(snowflake_ds).get_table_names(
+            schema=schema
+        )
```
Comment (Kilo59) [inline tests/datasource/fluent/integration/test_sql_datasources.py:323]: The `sqla.inspect()` command doesn't have return types defined and so I didn't notice that `get_table_names()` can take a `schema` param.

File: tests/datasource/fluent/integration/test_sql_datasources.py [hunk 2]
```diff
@@ -296,15 +335,20 @@ def test_snowflake(
             param(
                 "snowflake", get_random_identifier_name(), marks=[pytest.mark.snowflake]
             ),
+            param(
+                "databricks_sql",
+                PYTHON_VERSION,
+                marks=[pytest.mark.databricks],
```
Comment (Kilo59) [inline tests/datasource/fluent/integration/test_sql_datasources.py]: This particular test will continue to fail until I make the changes to how the `gx ExecutionEngine` is created.
Depending on when I get that done I may include that fix in this PR or skip this test and do it separately.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -275,6 +275,9 @@ jobs:
       AWS_ACCESS_KEY_ID: ${{secrets.AWS_ACCESS_KEY_ID}}
       AWS_DEFAULT_REGION: ${{secrets.AWS_DEFAULT_REGION}}
       AWS_SECRET_ACCESS_KEY: ${{secrets.AWS_SECRET_ACCESS_KEY}}
+      DATABRICKS_TOKEN: ${{secrets.DATABRICKS_SQL_TOKEN}}
+      DATABRICKS_HOST: ${{secrets.DATABRICKS_HOST}}
+      DATABRICKS_HTTP_PATH: ${{secrets.DATABRICKS_HTTP_PATH}}
```
Comment (Kilo59) [inline .github/workflows/ci.yml:279]: Still need these creds to be created and then added to github secrets.

File: reqs/requirements-dev-lite.txt [hunk 4]
```diff
@@ -18,4 +18,4 @@ pytest-xdist>=3.3.1
 requirements-parser>=0.2.0
 responses>=0.23.1 # requests mocking
 snapshottest==0.6.0 # GX Cloud atomic renderer tests
-sqlalchemy>=1.4.0
+sqlalchemy>=1.4.0,<2.0.0
```
Comment (Kilo59) [inline reqs/requirements-dev-lite.txt]:
```suggestion
sqlalchemy>=1.4.0
```

**Review by wookasz — APPROVED** [review]

**Pull request merged**